pkg/autoscaler/autoscaler.go

```diff
@@ -310,15 +311,22 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	observedPanicConcurrency := panicData.observedConcurrency(now)
 	observedStableConcurrencyPerPod := stableData.observedConcurrencyPerPod(now)
 	observedPanicConcurrencyPerPod := panicData.observedConcurrencyPerPod(now)
+
+	readyPods, err := a.readyPods(now)
+	if err != nil {
+		logger.Errorw("Failed to get Endpoints via K8S Lister", zap.Error(err))
+		return 0, false
+	}
```

Comment:
Should we move this before any computation above? If we want to exit early on a failure here, we might as well not do anything.

Reply:
Make sense. Done.